packages/stream/get-stream/get-stream.test.js

```diff
@@ -0,0 +1,132 @@
+      try {
+        await getStream('1')
+      } catch (error) {
+        expect(error).to.be.an('error')
+        expect(error.name).to.be.equal('TypeError')
+        expect(error.message).to.be.equal(
+          'Failed to get the stream data because the stream ID is not in number format. A stream ID must be number format'
+        )
+      }
+
+      try {
+        await getStream({})
+      } catch (error) {
+        expect(error).to.be.an('error')
+        expect(error.name).to.be.equal('TypeError')
+        expect(error.message).to.be.equal(
+          'Failed to get the stream data because the stream ID is not in number format. A stream ID must be number format'
+        )
+      }
+
+      try {
+        await getStream([])
+      } catch (error) {
+        expect(error).to.be.an('error')
+        expect(error.name).to.be.equal('TypeError')
+        expect(error.message).to.be.equal(
+          'Failed to get the stream data because the stream ID is not in number format. A stream ID must be number format'
+        )
+      }
+    })
+
+    it('should return error response if stream ID not found', async function () {
+      let stream_id = 1000
+      nock(`${Internal.config.api.base_url}`)
+        .get(`/${Internal.config.api.version}/streams/${stream_id}`)
+        .reply(404, { code: 404, message: 'ID not found', data: '' })
+
+      try {
+        await getStream(stream_id)
+      } catch (error) {
+        expect(error).to.be.an('error')
+        expect(error.name).to.be.equal('Error')
+        expect(error.message).to.be.equal(
+          'Failed to get the stream data because the stream ID is not found. Please provide a valid stream ID.'
+        )
+      }
+    })
+  })
+
+  describe('Positive test', function () {
+    let stream_id = 351
+    beforeEach(function () {
+      nock(`${Internal.config.api.base_url}`)
+        .get(`/${Internal.config.api.version}/streams/${stream_id}`)
+        .reply(200, {
+          status: {
+            code: 200,
+            message: 'Success',
+            type: 'success',
+          },
+          data: {
+            id: stream_id,
+            name: 'a new stream',
+            slug: '',
+            hls_manifest_path: '',
+            dash_manifest_path: '',
+            description: '',
+            created_by: 6,
+            created_at: '2022-09-05T07:38:09.525329768Z',
+            updated_at: '2022-09-05T07:38:09.525329768Z',
+            quality: '360',
+            viewer_count: 0,
+          },
+        })
+    })
+
+    afterEach(function () {
+      return nock.cleanAll()
+    })
+
+    it('should call the getStream function', async function () {
+      const result = await getStream(stream_id)
+      // console.log(result)
+
+      expect(result.status.code).to.equal(200)
+    })
+
+    it('should return success response', async function () {
+      const result = await getStream(stream_id)
```

Comment:
I want to make sure that when the `getStream` is called here, it will not hit to the real stream API endpoint and only use the nock to mock the request right? One way to test if the nock already works properly is the test should be stay working fine if the connection is disconnected

Reply:
@faiq-naufal yups mas, I've done trial before when the connection is disconnected, the test will stay working fine with using the mock data from nock.